src/omerocrate/uploader.py

```diff
@@ -225,9 +224,30 @@ def find_images(self) -> Iterable[tuple[Identifier, Path]]:
             file_path = result['file_path']
             yield file_path, Path(urlparse(file_path).path)
 
+    def find_image_ids(self) -> Iterable[tuple[Identifier, int]]:
+        """
+        Finds images with existing OMERO image IDs that should be processed but not uploaded to
+        OMERO (imageID is specified and upload value is false). This is typically used for adding
+        segmentations to existing images.
+        Can be overridden to customize the query.
+
+        Returns:
+            Yields tuples of (image URI, image ID).
+        """
+        for result in self.select_many("""
+            SELECT ?file_path ?image_id
+            WHERE {
+                ?file_path a schema:MediaObject ;
+                    omerocrate:upload false ;
+                    ome:ImageID ?image_id .
+            }
+        """):
+            file_path = result['file_path']
+            yield file_path, int(result['image_id'])
+
     def find_segmentation_for_image(self, image_uri: Identifier) -> Path | None:
         """
-        Finds the segmentation file associated with a given image URI.
+        Finds the segmentation file associated with a given image URI to be uploaded to OMERO.
```

Comment:
Why change this? Won't this be used for existing images?

Reply:
I was experimenting with using the `upload` flag for this query, which didn't work, I'll revert it.